### DISCUSSION TITLE
Long polling with suspense and refetchInterval

### DISCUSSION THREAD

**Discussion by andrewnaeve** [TanStack/query#4499] [Q&A] [answered]
Currently, when I set refetchInterval with suspense, I only see the spinner for the initial load. 
Afterwards, no fallback is shown. I've tried to reset the query, but this doesn't cleanly integrate with the refetch interval without me implementing it somehow myself. How should I approach this?

**Comment by TkDodo** [comment, marked as answer]
suspense is only shown instead of a "hard" loading spinner, where you have no data. Once you have data, react-query will show you stale data and _not_ suspend anymore. You can look at `isFetching` to see if a refetch is going on, and either replace our data with a spinner, or show the spinner in addition to stale data.

**Reply by andrewnaeve** [reply]
Ok, thank you for the reply and insight. It would be neat to allow controlling under what circumstances suspense is triggered, so that I could long poll without manually managing my spinners again. Would you consider a pull request if I had the time, or does this not align with current principles / or need more discussion?

**Reply by TkDodo** [reply]
I'm not sure. suspense is about to change with the new `use` api. Maybe we won't even have a suspense flag, but expose the promises so that everyone can `use(promise)` in user-land. This hasn't been decided yet.

If we keep the suspense flag, it might make sense to extend it to the format that most other flags have, which is:

```
suspense: boolean | 'always' | (query) => boolean
```

to give more fine-grained control. I personally consider it bad ux to unmount the current data api and show a spinner _instead_, but it's a decision that maybe shouldn't be taken away from suspense users.
We had a PR in that direction that got eventually not merged:
- https://github.com/TanStack/query/pull/3935

`resetQueries` does what you want, but it's different for `refetchInterval`.
